File: HelpSoftware/Help/Help.EF/Entitys/WrongTranslation.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class WrongTranslation : IHelpEntity
  {
    [Key]
    public virtual int ID_WrongTranslation { get; set; }

    public virtual string NewText { get; set; }

    public virtual string WrongText { get; set; }

    public virtual User Author { get; set; }

    public virtual DateTime Created { get; set; }

    public virtual string Description { get; set; }

    [Timestamp]
    public virtual byte[ ] TimeStamp { get; set; }

    public string GetEntityName()
    {
      throw new NotImplementedException();
    }

    public int GetID()
    {
      return ID_WrongTranslation;
    }

    public bool GetWriteHistoryEntry()
    {
      return false;
    }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/WrongTranslation.ts
import { User } from "./EF";

export class WrongTranslation {
  ID_WrongTranslation: number;
  NewText: string;
  WrongText: string;
  Author: User;
  Created: Date;
  Description: string;

  deserialize(input): WrongTranslation {
    this.ID_WrongTranslation = input.ID_WrongTranslation;
    this.NewText = input.NewText;
    this.WrongText = input.WrongText;
    this.Author = new User().deserialize(input.Author);
    this.Created = input.Created;
    this.Description = input.Description;
    return this;
  }
}
